### DISCUSSION TITLE
@layer CSS rule

### DISCUSSION THREAD

**Discussion by edamascus** [facebook/stylex#267] [Q&A] [answered]
The single css file, which is generated on build in StyleX, includes the `@layer` rule. I checked facebook.com and instagram.com and both do not have this rule in the styles file. I know that this rule was used to to tackle specifity issues; but I want to ask if it is well supported in order to be used in production. Do you expect developers to take care of browser compatibility after the file is generated or will this break the way css rules are applied?

When adding a seperate `globals.css` file for css resets and general styles for some elements, testing the build will result in applying those resets *after* the rules of the StyleX's single file. To make things work as expected, a `@layer` rule should be added to the `globals.css` file with lower priority. Is this an expected behavior?

**Comment by nmn** [comment, marked as answer]
You can already choose to use the polyfill for `@layer` if needed. Most bundler plugins will accept a `useLayers` option which can be set to `false`.

> if it is well supported in order to be used in production

As [you can see on caniuse.com](https://caniuse.com/?search=CSS%20Cascade%20Layers), CSS Cascade Layers have over 93% browser support. You have to decide if this is good enough for you and how you want to tackle that. You can ship the polyfill to all users, or generate two separate bundles. The main benefit of using `@layer` instead of the polyfill is that you get slightly small CSS files.

>  To make things work as expected, a @layer rule should be added to the globals.css file with lower priority. Is this an expected behavior?

If you choose to use `@layer`, yes this is expected behavior.
